### ISSUE TITLE
[Bug]  Assert triggered on some CPUs

### ISSUE THREAD

**Issue by irisitystefannilsson** [openvinotoolkit/openvino#9877]
##### System information (version)
- OpenVINO => master (2022-01-24)
- Operating System / Platform => Windows 10 64 Bit
- Compiler => Visual Studio 2019
- CPU=> Intel(R) Core(TM) i7-11800H @ 2.30GHz

##### Detailed description
When running inference from c++ code built for Debug an assert is triggered saying:

Assertion failed: IMPLICATION(!(is_zero_point || jcp.with_input_zp), idx == ker_dw_reg_base_idx), file C:\Users\StefanNilsson\source\repos\openvino\src\plugins\intel_cpu\thirdparty\mkl-dnn\src\cpu\x64\jit_avx512_core_x8s8s32x_conv_kernel.cpp, line 967

This only happens on some machines/cpus. My  team members with somewhat older machines have not seen this.

Running code built for Release seems to work (produces no error messages), but I have not checked the results for correctness on my machine.
 I first observed the problem in 2021.4.1.

**Comment by Iffa-Meah** [comment]
Hi @irisitystefannilsson ,

Please help to provide these:
1. The c++ code built for Debug?
2. Which OpenVINO inference app did you run this debug code with?
3. Command to run these code
4. Any related model files in order to test this from our end

**Comment by irisitystefannilsson** [comment]
2. I was able to trigger the assert when running the 'hello_classification' sample code in : 
openvino/samples/cpp/hello_classification/
3. ./hello_classification objectdetector.xml img.bmp CPU  
4. The 'objectdetector'  is an in-house model with input shape [1, 3, 96, 96]. I will see if I can upload something similar that triggers the asssert

I was able to reproduce it on linux/gcc as well (ubuntu 20.04 on wsl2), so it does not seem to be compiler dependent, only seemingly cpu dependent

**Comment by Iffa-Meah** [comment]
@irisitystefannilsson ,

I don't see any issue from my side. I'm using Windows 10 with Intel Core i7 CPU. I cloned the latest master branch.

<img width="669" alt="w10BuildFromSrc" src="https://user-images.githubusercontent.com/79643476/151582303-96ac7011-93ad-4fff-a014-86eb10b4676a.png">


**Make sure to follow these steps:**

1. Clone the latest repo: git clone https://github.com/openvinotoolkit/openvino.git
2. Navigate to the cloned folder: cd openvino
3. Navigate around git repo: git checkout tags/2021.2
4. Sets the Git repository of the submodule to this particular commit: git submodule update --init --recursive
5. Create new dir: mkdir build
6. Navigate to dir: cd build
7. Compile: cmake -G "Visual Studio 16 2016" -A x64 -DCMAKE_BUILD_TYPE=Release ..
8. Build: cmake --build . --config Release --verbose -j8

Once everything is done, add tbb and opencv dir location to PATH:
<img width="544" alt="pathtbb" src="https://user-images.githubusercontent.com/79643476/151580395-24fdabf7-87ba-4946-9ccf-b4e0f5d2006f.png">

If the default OpenVINO sample with supported model actually works (especially in your system) this means **there's something wrong in your c++ code built for Debug**. You'll need to review & debug the code itself.

For the Model Optimizer, pay more attention to the [When to Specify Input Shapes](https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_Converting_Model.html#when-to-specify-input-shapes) (--input_shape) and [When to Reverse Input Channels](https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_Converting_Model.html#when-to-reverse-input-channels) (--reverse_input_channels), which you can refer [here](https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_Converting_Model.html).

**Comment by Iffa-Meah** [comment]
The OpenVINO CPU plugin does supports inference on Intel® Xeon® with Intel® Advanced Vector Extensions 2 (Intel® AVX2), Intel® Advanced Vector Extensions 512 (Intel® AVX-512), and AVX512_BF16, Intel® Core™ Processors with Intel® AVX2, Intel Atom® Processors with Intel® Streaming SIMD Extensions (Intel® SSE).

In OpenVINO build from source, ENABLE_AVX512F parameter would enables AVX512 optimizations. It's ON by default for x86 platforms, but **not available for other platforms**.  This would affect only the Inference Engine common part and preprocessing plugin, does not affect the mkldnn library.

Using default configurations, I don't see any issue from my side.



<img width="677" alt="HELLO_CLASSIFICATION" src="https://user-images.githubusercontent.com/79643476/152027498-ad6e5998-9cc3-42a8-ac7e-06917c93bca2.png">


However, we'll look further into this.

**Comment by irisitystefannilsson** [comment]
If the OpenVINO is built for Release the Assert will probably not be checked at runtime. 
I only see this if OpenVINO is built for Debug.

**Comment by Iffa-Meah** [comment]
We are investigating this on the 11th gen system. We'll revert back to you asap.

**Comment by jgespino** [comment]
Hi @irisitystefannilsson 

I am seeing the same behavior on my system with 11th gen Intel Core processor. I have reached out to the development team for additional assistance. I will let you know what I find out. 

Built OpenVINO as Debug on 9th Gen Intel Core Processor:
```
~/inference_engine_build_samples_build/intel64/Release/hello_classification ~/Downloads/frozen_inference_graph_int8/optimized/frozen_inference_graph.xml ~/Downloads/w3c_home.bmp CPU
[ INFO ] OpenVINO Runtime version ......... 2022.1.0
[ INFO ] Build ........... custom_master_abeb910ce27831fad866200852b5b1aa0dce4803
[ INFO ] 
[ INFO ] Loading model files: /home/pse/Downloads/frozen_inference_graph_int8/optimized/frozen_inference_graph.xml
[ INFO ] model name: frozen_inference_graph
[ INFO ]     inputs
[ INFO ]         input name: image_tensor
[ INFO ]         input type: f32
[ INFO ]         input shape: {1, 3, 300, 300}
[ INFO ]     outputs
[ INFO ]         output name: DetectionOutput
[ INFO ]         output type: f32
[ INFO ]         output shape: {1, 1, 100, 7}

Top 10 results:

Image /home/pse/Downloads/w3c_home.bmp

classid probability
------- -----------
8       0.0000000  
9       0.0000000  
4       0.0000000  
7       0.0000000  
5       0.0000000  
3       0.0000000  
1       0.0000000  
10      0.0000000  
2       0.0000000  
6       0.0000000 
```

Built OpenVINO as Debug on 11th Gen Intel Core Processor:
```
~/inference_engine_build_samples_build/intel64/Release/hello_classification ~/Downloads/frozen_inference_graph_int8/optimized/frozen_inference_graph.xml ~/Downloads/w3c_home.bmp CPU
[ INFO ] OpenVINO Runtime version ......... 2022.1.0
[ INFO ] Build ........... custom_master_317b956d2e1b531a88f1bf8ffb32430290f28a32
[ INFO ] 
[ INFO ] Loading model files: frozen_inference_graph.xml
[ INFO ] model name: frozen_inference_graph
[ INFO ]     inputs
[ INFO ]         input name: image_tensor
[ INFO ]         input type: f32
[ INFO ]         input shape: {1, 3, 300, 300}
[ INFO ]     outputs
[ INFO ]         output name: DetectionOutput
[ INFO ]         output type: f32
[ INFO ]         output shape: {1, 1, 100, 7}
hello_classification: /home/pse/Downloads/openvino/src/plugins/intel_cpu/thirdparty/mkl-dnn/src/cpu/x64/jit_avx512_core_x8s8s32x_conv_kernel.cpp:981: void dnnl::impl::cpu::x64::_jit_avx512_core_x8s8s32x_fwd_kernel<Vmm>::generate() [with Vmm = Xbyak::Zmm]: Assertion `IMPLICATION(!(is_zero_point || jcp.with_input_zp), idx == ker_dw_reg_base_idx)' failed.
Aborted (core dumped)
```

Regards,
Jesus

Ref. 79577

**Comment by jgespino** [comment]
@irisitystefannilsson This has been fixed in the [2022.2-dev branch](https://github.com/openvinotoolkit/openvino/tree/2022.2.0.dev20220829). Could you please test from your end and let me know if it works for you too?

```
~/inference_engine_build_samples_build_2022.1/intel64/Release$ ./hello_classification ~/frozen_inference_graph.xml ~/image.bmp CPU
[ INFO ] OpenVINO Runtime version ......... 2022.1.0
[ INFO ] Build ........... custom_HEAD_cdb9bec7210f8c24fde3e416c7ada820faaaa23e
[ INFO ] 
[ INFO ] Loading model files: /home/pse/frozen_inference_graph.xml
[ INFO ] model name: frozen_inference_graph
[ INFO ]     inputs
[ INFO ]         input name: image_tensor
[ INFO ]         input type: f32
[ INFO ]         input shape: {1, 3, 300, 300}
[ INFO ]     outputs
[ INFO ]         output name: DetectionOutput
[ INFO ]         output type: f32
[ INFO ]         output shape: {1, 1, 100, 7}
hello_classification: /home/pse/Downloads/openvino-2022.1/src/plugins/intel_cpu/thirdparty/mkl-dnn/src/cpu/x64/jit_avx512_core_x8s8s32x_conv_kernel.cpp:981: void dnnl::impl::cpu::x64::_jit_avx512_core_x8s8s32x_fwd_kernel<Vmm>::generate() [with Vmm = Xbyak::Zmm]: Assertion `IMPLICATION(!(is_zero_point || jcp.with_input_zp), idx == ker_dw_reg_base_idx)' failed.
Aborted (core dumped)
```

```
~/inference_engine_cpp_samples_build_2022.2-dev/intel64/Release$ ./hello_classification ~/frozen_inference_graph.xml ~/image.bmp CPU
[ INFO ] OpenVINO Runtime version ......... 2022.2.0
[ INFO ] Build ........... custom_HEAD_41a404f2908be7bcf7c7c7f588d8ef389366a018
[ INFO ] 
[ INFO ] Loading model files: /home/pse/frozen_inference_graph.xml
[ INFO ] model name: frozen_inference_graph
[ INFO ]     inputs
[ INFO ]         input name: image_tensor
[ INFO ]         input type: f32
[ INFO ]         input shape: {1, 3, 300, 300}
[ INFO ]     outputs
[ INFO ]         output name: DetectionOutput
[ INFO ]         output type: f32
[ INFO ]         output shape: {1, 1, 100, 7}

Top 10 results:

Image /home/pse/image.bmp

classid probability
------- -----------
8       3.0000000  
1       3.0000000  
12      0.9998221  
2       0.9879607  
13      0.9808570  
10      0.8560728  
6       0.7522659  
5       0.6641450  
11      0.3972979  
3       0.3071242 
```
